tests/benchmark/README.md

```diff
@@ -0,0 +1,320 @@
+# Vector Similarity benchmarks
+
+## Table of contents
+* [Overview](#overview)
+* [Run benchmarks](#run-benchmarks)
+* [Available sets](#available-sets)  
+    - [BM_VecSimBasics](#bm_vecsimbasics)  
+    - [BM_BatchIterator](#bm_batchiterator)  
+    - [BM_VecSimUpdatedIndex](#bm_vecsimupdatedindex)  
+* [The benchmarks directory](#the-benchmarks-directory)
+    - [How to define and register a new test](#how-to-define-and-register-a-new-test)
+* [Fixtures](#fixtures)
+    - [Benchmarks fixture classes hierarchy](#benchmarks-fixture-classes-hierarchy)
+* [Cmake remarks](#cmake-remarks)
+* [ann-benchmarks](#ann-benchmark)
+
+# Overview
+We use Google benchmark tool to create our benchmarks.
+Google benchmark is a popular tool for benchmark code snippets, similar to unit tests. It allows a quick way of estimating the runtime of each test based on several (identical) runs, and print the results as output. For some tests, the output includes additional "Recall" metric, which indicates the accuracy in case of approximate search.  
+**The recall** is calculated as the number of results from the approximate algorithm (HNSW in this case) divided by the ground truth results number (calculated by the brute force algorithm).
+
+# Run benchmarks
+## Required files
+The serialized hnsw indices and the datasets for the ann-benchmarks can be found in 
+`tests/benchmark/data/hnsw_indices.txt`.  
+To download all the required files run:
+```asm
+wget -q -i tests/benchmark/data/hnsw_indices.txt -P tests/benchmark/data
+```
+To run all test sets, call the following commands from the project root dir:
+```asm
+make benchmark
+```
+### Running a Subset of Benchmarks
+To run only a subset of benchmarks that match a specified `<regex>`, set `BENCHMARK_FILTER=<regex>` environment variable. For example:  
+```asm
+make benchmark BENCHMARK_FILTER=fp32*
+```
+
+# Available sets
+There are currently 3 sets of benchmarks available: `BM_VecSimBasics` , `BM_BatchIterator` and `BM_VecSimUpdatedIndex`. Each is templated according to the index data type.
+## BM_VecSimBasics
+For each combination of data type (fp32/fp64) and label type (single/multi) the following test cases are included:
+1. Index `total memory` 
+2. `AddLabel` - adds `DEFAULT_BLOCK_SIZE (= 1024)` new labels to the index from the test_query_file. Note that for a single value index each label contains one vector, meaning that the number of new labels equals the number of new vectors.  
```

Comment:
i didn't mention any specific attributes of the datasets to leave the option to change them in the future.
I'll add a section describing all the current datasets' parameters so it will be written in one place and easy to change if needed.